// js-dataverse/dist/utils/datasetUtil.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
var metadataBlocksUtil_1 = require("./metadataBlocksUtil");
exports.EMPTY_ARRAY_LENGTH = 0;
var DatasetUtil = /** @class */ (function () {
    function DatasetUtil() {
    }
    DatasetUtil.mapBasicDatasetInformation = function (datasetInformation) {
        var fields = [];
        if (!datasetInformation) {
            return undefined;
        }
        if (datasetInformation.title) {
            fields.push(metadataBlocksUtil_1.MetadataBlocksUtil.createTitleField(datasetInformation.title));
        }
        if (datasetInformation.subtitle) {
            fields.push(metadataBlocksUtil_1.MetadataBlocksUtil.createSubtitleField(datasetInformation.subtitle));
        }
        if (datasetInformation.authors && datasetInformation.authors.length > exports.EMPTY_ARRAY_LENGTH) {
            fields.push(metadataBlocksUtil_1.MetadataBlocksUtil.createAuthorField(datasetInformation.authors));
        }
        if (datasetInformation.descriptions && datasetInformation.descriptions.length > exports.EMPTY_ARRAY_LENGTH) {
            fields.push(metadataBlocksUtil_1.MetadataBlocksUtil.createDescriptionField(datasetInformation.descriptions));
        }
        if (datasetInformation.subject) {
            fields.push(metadataBlocksUtil_1.MetadataBlocksUtil.createSubjectField(datasetInformation.subject));
        }
        if (datasetInformation.contact) {
            fields.push(metadataBlocksUtil_1.MetadataBlocksUtil.createContactField(datasetInformation.contact));
        }
        return {
            datasetVersion: {
                metadataBlocks: {
                    citation: {
                        fields: fields,
                        displayName: 'Citation Metadata'
                    }
                }
            }
        };
    };
    return DatasetUtil;
}());
exports.DatasetUtil = DatasetUtil;
